src/batch/src/executor/monitor/stats.rs

```diff
@@ -11,16 +14,146 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-//
-use prometheus::core::{AtomicU64, GenericCounterVec};
+use prometheus::core::{AtomicU64, Collector, GenericCounterVec};
 use prometheus::{
-    exponential_buckets, histogram_opts, register_histogram_with_registry,
+    exponential_buckets, histogram_opts, opts, register_histogram_with_registry,
     register_int_counter_vec_with_registry, Histogram, Registry,
 };
+use tokio::sync::mpsc::Sender;
+
+use crate::task::TaskId;
+
+// When execution is done, it need to call clear_record() in BatchTaskMetrics.
+// The clear_record() will send the Collector to delete_queue, if the queue is full, the execution
+// will be blocked so that user can't get the result immediately.
+const DELETE_QUEUE_SIZE: usize = 4096;
```

Comment:
I find that the execution will be blocked when the delete queue is full. And that will caused the execution time be longer(can't not pass e2e test). 
For now, I set it to 4096 and seems it can pass the e2e test. But we may need better solution later. @liurenjie1024 @BowenXiao1999

Reply:
I think another solution is to `select` futures of timers and deletion queue so that it will not block task clean up.

Reply:
Can we spawn a task to write the channel to avoid this?

Reply:
You can take this for example.

Reply:
> I think another solution is to select futures of timers and deletion queue so that it will not block task clean up.
It may can alleviate. But it can't avoid some case where the collector is too much?

> Can we spawn a task to write the channel to avoid this?
I'm afraid that this will cause too much task.

Reply:
I think we can make this queue unbounded, anyway it should exist in memory, so ubouned queue is ok to me. But we need to add a metric to monitor the pending tasks.